File: 프로그래머스/lv2/42842. 카펫/카펫.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    queue<pair<int,int>> divisor;
    
    //1. yellow의 약수 구하기
    if(yellow == 1) {
        divisor.push({1,1});
        
    }
    else {
        for(int i=1; i<=yellow/2; i++) {
            if(yellow % i == 0) {
                divisor.push({i,yellow/i});
            }
        }
    }
    while(!divisor.empty()) {
        int x = divisor.front().first;
        int y = divisor.front().second;
        divisor.pop();
        
        if(x + y == (brown - 4)/2) {
            if(x <= y) {
                answer.push_back(y+2);
                answer.push_back(x+2);
            }
            else {
                answer.push_back(x+2);
                answer.push_back(y+2);
                
            }
            break;
        }
        
    }
    
    
    return answer;
}

File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    var answer = [];
    var sum = brown + yellow;
    var w=0, h=0;
    for(let i=1; i<=brown; i++) {
        for(let j=brown; j>= 1; j--) {
            if(2*i + 2*j - 4 === brown && i*j === sum)  {
                answer.push(Math.max(i,j));
                answer.push(Math.min(i,j));
                return answer;
                
            }
        }
    }
    
}